dot/network/service.go

```diff
@@ -153,7 +153,10 @@ func (s *Service) handleConn(conn network.Conn) {
 	if !s.noStatus {
 
 		// get latest block header from block state
-		latestBlock := s.cfg.BlockState.LatestHeader()
+		latestBlock, err := s.cfg.BlockState.BestBlockHeader()
+		if err != nil {
+			log.Error("[network] could not get chain head", "err", err)
```

Comment:
since its returning a error, if we ignore it here, is there any changes that calling latestBlock would return a nillpointer ?

Reply:
will return if error